agent/xds/listeners.go

```diff
@@ -1990,6 +2016,7 @@ func makeTCPProxyFilter(filterName, cluster, statPrefix string) (*envoy_listener
 
 func makeConnectionLimitFilter(limit int) (*envoy_listener_v3.Filter, error) {
 	cfg := &envoy_connection_limit_v3.ConnectionLimit{
+		StatPrefix:     "inbound connection limit",
```

Comment:
This should have underscores for consistency with other stats prefixes:
```suggestion
		StatPrefix:     "inbound_connection_limit",
```